### PR TITLE
feat: bind social account

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Bind social account to an existing user.

image

## Linear Issue Reference

LOG-630

### CODE REVIEW

File: packages/core/src/lib/register.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { PasscodeType, UserLogType } from '@logto/schemas';
 import { Context } from 'koa';
-import { Provider } from 'oidc-provider';
+import { InteractionResults, Provider } from 'oidc-provider';
```
Comment (darcyYe) [inline packages/core/src/lib/register.ts:3]: imported but not used?

File: packages/core/src/routes/session.ts [hunk 2]
```diff
@@ -228,6 +230,35 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
+    koaGuard({
+      body: object({
+        connectorId: string(),
+      }),
+    }),
+    async (ctx, next) => {
+      const { connectorId } = ctx.guard.body;
+      const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+      assertThat(result, 'session.connector_session_not_found');
+      assertThat(result.login?.account, 'session.unauthorized');
```
Comment (darcyYe) [inline packages/core/src/routes/session.ts:244]: not sure whether this should be the following code snippet?
```suggestion
      assertThat(result.login?.accountId, 'session.unauthorized');
```

FYI
image
image

File: packages/phrases/src/locales/en.ts [hunk 3]
```diff
@@ -66,6 +66,7 @@ const errors = {
     insufficient_info: 'Insufficent sign-in info.',
     connector_id_mismatch: 'The connectorId is mismatched with session record.',
     connector_session_not_found: 'Connector session not found. Please go back and sign in again.',
+    unauthorized: 'Please sign in first.',
```
Comment (darcyYe) [inline packages/phrases/src/locales/en.ts:69]: nit: please sign in before binding?

**Review by darcyYe — CHANGES_REQUESTED** [review]

File: packages/phrases/src/locales/en.ts (hunk 3 above)
Comment (wangsijie) [reply packages/phrases/src/locales/en.ts:69]: Other actions may require sign in as well.

**Review by darcyYe — APPROVED** [review]
lgtm

**Pull request merged**